### ISSUE TITLE
0.9.5 KeyError: 'readOnly'

### ISSUE THREAD

**Issue by jayvdb** [tfranzel/drf-spectacular#59]
Commented out the fixes from https://gist.github.com/tfranzel/c29f6b79967c6d4b89b78e779e2a798f , and a different backtrace appears

```py
Traceback (most recent call last):
  File "./manage.py", line 46, in <module>
    main()
  File "./manage.py", line 42, in main
    execute_from_command_line(sys.argv)
  File "/usr/lib/python3.8/site-packages/django/core/management/__init__.py", line 381, in execute_from_command_line
    utility.execute()
  File "/usr/lib/python3.8/site-packages/django/core/management/__init__.py", line 375, in execute
    self.fetch_command(subcommand).run_from_argv(self.argv)
  File "/usr/lib/python3.8/site-packages/django/core/management/base.py", line 323, in run_from_argv
    self.execute(*args, **cmd_options)
  File "/usr/lib/python3.8/site-packages/django/core/management/base.py", line 364, in execute
    output = self.handle(*args, **options)
  File "/usr/lib/python3.8/site-packages/drf_spectacular/management/commands/spectacular.py", line 44, in handle
    schema = generator.get_schema(request=None, public=True)
  File "/usr/lib/python3.8/site-packages/drf_spectacular/generators.py", line 153, in get_schema
    paths=self.parse(request, public),
  File "/usr/lib/python3.8/site-packages/drf_spectacular/generators.py", line 133, in parse
    operation = view.schema.get_operation(path, path_regex, method, self.registry)
  File "/usr/lib/python3.8/site-packages/drf_spectacular/openapi.py", line 77, in get_operation
    operation['responses'] = self._get_response_bodies()
  File "/usr/lib/python3.8/site-packages/drf_spectacular/openapi.py", line 816, in _get_response_bodies
    return {'200': self._get_response_for_code(response_serializers)}
  File "/usr/lib/python3.8/site-packages/drf_spectacular/openapi.py", line 841, in _get_response_for_code
    component = self.resolve_serializer(serializer, 'response')
  File "/usr/lib/python3.8/site-packages/drf_spectacular/openapi.py", line 913, in resolve_serializer
    component.schema = self._map_serializer(serializer, direction)
  File "/usr/lib/python3.8/site-packages/drf_spectacular/openapi.py", line 582, in _map_serializer
    schema = self._map_basic_serializer(serializer, direction)
  File "/usr/lib/python3.8/site-packages/drf_spectacular/openapi.py", line 636, in _map_basic_serializer
    schema = self._map_serializer_field(field, direction)
  File "/usr/lib/python3.8/site-packages/drf_spectacular/openapi.py", line 419, in _map_serializer_field
    schema = self.resolve_serializer(field.child, direction).ref
  File "/usr/lib/python3.8/site-packages/drf_spectacular/openapi.py", line 913, in resolve_serializer
    component.schema = self._map_serializer(serializer, direction)
  File "/usr/lib/python3.8/site-packages/drf_spectacular/openapi.py", line 582, in _map_serializer
    schema = self._map_basic_serializer(serializer, direction)
  File "/usr/lib/python3.8/site-packages/drf_spectacular/openapi.py", line 636, in _map_basic_serializer
    schema = self._map_serializer_field(field, direction)
  File "/usr/lib/python3.8/site-packages/drf_spectacular/openapi.py", line 437, in _map_serializer_field
    del schema['readOnly']
KeyError: 'readOnly'
```

I'll get some better stack information shortly to identify where this and the other problem are coming from.

**Comment by jayvdb** [comment]
schema: `{'type': 'string'}`

field: `PrimaryKeyRelatedField(queryset=Country.objects.all())`

probably [`CountryList`](https://github.com/django-oscar/django-oscar-api/blob/5628e171422f5de06f6f48c174239c119f7f792d/oscarapi/views/basic.py#L52)

**Comment by tfranzel** [comment]
that is interesting. i heavily tested this and never encoutered this case. i was under the assumption that this was always true. apparently not. easily fixed. 

https://github.com/tfranzel/drf-spectacular/blob/f2f87683fe0f43879cbd1c2b369443f83fa8a845/drf_spectacular/openapi.py#L435-L437

**Comment by jayvdb** [comment]
`schema.pop('readOnly', None)` fixes this for me.

**Comment by tfranzel** [comment]
fixed by 5d15ef1
